picard/webservice/__init__.py

```diff
@@ -223,46 +233,51 @@ def _handle_reply(self, reply, request, handler, parse_response_format, refresh)
                         self.get(redirect_host,
                                  redirect_port,
                                  redirect_path,
-                                 handler, parse_response_format, priority=True, important=True, refresh=refresh, queryargs=redirect_query,
+                                 handler, parse_response_type, priority=True, important=True, refresh=refresh, queryargs=redirect_query,
                                  cacheloadcontrol=request.attribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute))
                     else:
                         log.error("Redirect loop: %s",
                                   reply.request().url().toString(QUrl.RemoveUserInfo)
                                   )
                         handler(reply.readAll(), reply, error)
-                elif parse_response_format:
-                    response_parser = self.response_parser(parse_response_format)
+                elif parse_response_type:
                     try:
-                        document = response_parser(reply)
+                        response_parser = self.get_response_parser(parse_response_type)
                     except UnknownResponseParserError as e:
-                        log.error("Attempting to parse the response body without a proper parser")
+                        log.error(e.args[0])
                         document = reply.readAll()
+                    else:
+                        try:
+                            document = response_parser(reply)
+                        except TypeError as e:
```

Comment:
If an exception other than a TypeError is raised, document will still be undefined later on. It's also not entirely clear to me why TypeError is special here.